### DISCUSSION TITLE
Issue with DMD Compiler

### DISCUSSION THREAD

**Discussion by csantosbntt** [abraunegg/onedrive#2481] [Q&A] [answered]
Hello! I'm experiencing an issue while updating OneDrive.

I have downloaded the Git repository with the latest version and am following the update steps. However, when I run the following command, I encounter the following error:

![Captura2](https://github.com/abraunegg/onedrive/assets/140811510/553091d7-ee90-4993-8e20-2731d6ba9e1e)

Upon analysis, I've realized that the issue is because the DMD package is not installed on my server. The security department has advised against installing it since it's not available in the official CentOS repository, and they do not recommend installing unofficial repositories for security reasons.

Is there anything I can do to resolve this problem?

Thank you very much.

**Comment by abraunegg** [comment, marked as answer]
@csantosbntt 
The 'onedrive' client is installable through EPEL for CentOS platforms:

![image](https://github.com/abraunegg/onedrive/assets/4956234/2b606152-03b2-40af-b4e6-edb6d085d41d)

My advice is that you use the EPEL repository to install rather than self compiling.

LDC is also available via EPEL to allow you to build the client using official packages - however this all depends on if the EPEL repository is considered 'official' by your University.

Generally, not installing packages from unknown publishers should be the advice your University should be advising. Installing packages from known publishers from their official site should be an acceptable position if the provided security validation of the package to install matches. You can install LDC for CentOS using the official packages from here: https://dlang.org/download.html

**Comment by csantosbntt** [comment]
Okey, I will contact the security team and ask them about EPL. Thank you very much.

Regards!
